### PR TITLE
enable smooth environment variable naming transition

### PR DESCRIPTION
**Pull request by MoralCode**
**Description**
This PR introduces a new `SystemEnv` class that lives alongside `SystemConfig` and aims to centralize/unify all environment variable access.

The primary purpose of this is to slightly decouple the requested environment variable name from the name where the value is actually pulled from. For example, even if all in-code references use `COLLECTOSS_DB` for the database connection string, this layer will allow a values to be read from `AUGUR_DB` as a fallback if `COLLECTOSS_DB` (preferred) is not set.

This class is also set up to print a deprecation warning/notice whenever the `AUGUR` variant of an environment variable is used as a fallback. The message makes it clear that this fallback behavior is only temporary and may not be around forever (although I don't see this code being removed anytime in the foreseeable future)

This PR includes a unit tests for this new functionality and works as intended, including edge cases such as when the key the code requests doesn't use a known prefix (`os.getenv("CONFIG_DATADIR")`)

It also includes a special case function `get_bool()` because there were ~3 different places in the code where a small python one-liner (that i wrote) had been copied around for convenience, and this was a good opportunity to refactor that to prevent those three versions from ever drifting in implementation from each other. There are certainly more places this logic can be reused as well as we unify the way all env vars are processed.

Environment variables handled via Click (the CLI library) are addressed too. the click automatic prefix has been updated and the ~5 lines in 1 file where we were explicitly setting the variables are now updated to include both variants.

This also addresses the init, config, and entrypoint bash scripts uses at startup by replacing them with python equivalents (and enabling them to utilize the new environment variable switcher as well). This effectively means the earliest phases of the startup sequence in CollectOSS have been cleaned up somewhat

This PR resolves one of the large items that needs to be done to complete #279 for the next release.

**Notes for Reviewers**
This change renames all code references to use the new `COLLECTOSS_` prefix, with the `SystemEnv` adapting these accesses to be able to read from `AUGUR_` variables if `COLLECTOSS_` ones aren't present.

This PR works in the sense that CollectOSS is able to start up with this change present with the default env

**Signed commits**
- [X] Yes, I signed my commits.

### CODE REVIEW

File: collectoss/tasks/init/celery_app.py
Comment (shlokgilda) [inline collectoss/tasks/init/celery_app.py:1]: This file is missing the `SystemEnv` import.

File: collectoss/application/environment.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        str: The detected prefix (including any separators) if any, otherwise None
+    """
+    prefix_len = 0
+    for p in prefixes:
+        p = p.upper()
+        k = key.upper()
+        if k.startswith(p):
+            prefix_len += len(p)
+
+            if k[prefix_len] == separator:
```
Comment (shlokgilda) [inline collectoss/application/environment.py]: Two bugs in this block, easier to fix together:
* `k[prefix_len]` will raise an error (likely IndexError) when the key is exactly the prefix (e.g. `extract_prefix("AUGUR", ["AUGUR"])`).
* The `return key[0:prefix_len]` is unconditional once `startswith` matches, so `extract_prefix("AUGURDB", ["AUGUR"])` returns `"AUGUR"` even though `AUGURDB` isn't actually prefixed. That then makes `SystemEnv.get("AUGURDB")` look up `COLLECTOSS_DB`/`AUGUR_DB`, which I think is wrong?

File: collectoss/application/environment.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                prefix_len += len(separator)
+            return key[0:prefix_len]
+    return None
```
Comment (shlokgilda) [inline collectoss/application/environment.py]:
```suggestion
    k = key.upper()
    for p in prefixes:
        p_up = p.upper()
        if k == p_up:
            return key[:len(p)]
        if k.startswith(p_up + separator):
            return key[:len(p) + len(separator)]
    return None
```
Should probably also update the docstrings and and worth adding tests for both: `extract_prefix("AUGUR", ...)` and `extract_prefix("AUGURDB", ...)`?

File: collectoss/application/environment.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            check_key = f"{p}{cls._separator}{suffix}"
+            value = os.getenv(check_key, None)
+
+            if value is not None:
+                # emit a warning if configured
+                if p in cls._warn_prefixes:
+                    msg = (
+                        f"Environment variable '{check_key}' is deprecated. "
+                        f"Use '{key}' instead. This automatic recovery may become a failure in a future version "
+                    )
```
Comment (shlokgilda) [inline collectoss/application/environment.py:54]: Isn't this deprecation message telling the user to use the variable they already used? When call sites are still `SystemEnv.get("AUGUR_DB")`, `key` *is* `AUGUR_DB`, so the warning reads: "Environment variable 'AUGUR_DB' is deprecated. Use 'AUGUR_DB' instead." Should point at the canonical (non-warn) name?

File: collectoss/application/environment.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            return os.getenv(key, default)
+        
+        return default
+    
+    @classmethod
+    def get_bool(cls, key:str, default: bool, prefixes = _prefixes) -> bool:
+        """gets a value from the environment and cast it to a boolean
+        """
+        raw_val = cls.get(key, None, prefixes)
+        return raw_val.lower() in ('true', '1', 't', 'y', 'yes') if raw_val else default
```
Comment (shlokgilda) [inline collectoss/application/environment.py]: `if raw_val` is falsy on an empty string, so `FOO=""` returns `default` instead of `False`. Compared to the old `os.getenv("FOO", "False").lower() in (...)` pattern, it actually flips outcomes for the cases where `default=True`.

```suggestion
        raw_val = cls.get(key, None, prefixes)
        if raw_val is None:
            return default
        return raw_val.lower() in ('true', '1', 't', 'y', 'yes')
```

File: collectoss/application/environment.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        
+    @classmethod
+    def set(cls, key: str, value: str, overwrite=True) -> None:
```
Comment (shlokgilda) [inline collectoss/application/environment.py:75]: `set` writes the literal key, but `get` does prefix translation. so `SystemEnv.set("AUGUR_DEV", "1")` in `backend.py` plants an `AUGUR_DEV` that the next `SystemEnv.get(...)` happily resolves via fallback and warns about. The codebase ends up emitting deprecation warnings against itself. Similar to the deprecation warning issue earlier in `get` method. Let's have `set` rewrite to the canonical prefix (so writes always land on `COLLECTOSS_*`)?

File: collectoss/tasks/git/dependency_tasks/core.py [hunk 6]
```diff
@@ -79,19 +80,11 @@ def generate_scorecard(logger, repo_git):
     command = '--repo=' + path
     
     #this is path where our scorecard project is located
-    path_to_scorecard = os.getenv('SCORECARD_DIR', os.environ['HOME'] + '/scorecard')
+    path_to_scorecard = SystemEnv.get('SCORECARD_DIR', (SystemEnv.get('HOME') or "~") + '/scorecard')
```
Comment (shlokgilda) [inline collectoss/tasks/git/dependency_tasks/core.py]: Shouldn't this be:

```suggestion
    path_to_scorecard = SystemEnv.get('SCORECARD_DIR', os.path.expanduser('~/scorecard'))
```

If `HOME` is unset, the fallback becomes the literal string `~/scorecard`. `subprocess`'s `cwd=` won't expand `~` (that's a shell thing), so this silently hands a broken path downstream. The old `os.environ['HOME']` would `KeyError` loudly. Wouldn't that be better?


Same thing applies to `SCC_DIR` in `collectoss/tasks/git/scc_value_tasks/core.py:24`.

**Review by shlokgilda — CHANGES_REQUESTED** [review]
Added some inline comments

File: collectoss/application/environment.py (hunk 1 above)
Comment (MoralCode) [reply collectoss/application/environment.py]: good catches!

File: collectoss/application/environment.py (hunk 5 above)
Comment (MoralCode) [reply collectoss/application/environment.py:75]: This is intentional, the next step of this PR, ideally before the next release, is to change all the env vars to use the COLLECTOSS_ prefix

Comment (MoralCode) [reply collectoss/application/environment.py:75]: maybe this is a misguided attempt to not do too much in one PR lol

File: collectoss/tasks/git/dependency_tasks/core.py (hunk 6 above)
Comment (MoralCode) [reply collectoss/tasks/git/dependency_tasks/core.py]: is `.expanduser()` influenced by `HOME`?

File: collectoss/application/environment.py (hunk 3 above)
Comment (MoralCode) [reply collectoss/application/environment.py:54]: yes, this should automatically change when we move the call sites over to the new name

Comment (MoralCode) [reply collectoss/application/environment.py:54]: which i might just lump into this PR

File: collectoss/util/startup.py [hunk 7]
```diff
@@ -0,0 +1,244 @@
+## Startup helpers
+
+
+from pathlib import Path
+import os
+import getpass
+import subprocess
```
[the remark this answers is not in the document]
Comment (MoralCode) [reply collectoss/util/startup.py:7]: yep, this is going to be removed soon
